slam_toolbox:
  ros__parameters:
    use_sim_time: true

    
    map_frame: map
    odom_frame: world
    base_frame: boat_1/hull_1
    scan_topic: /scan

    
    mode: mapping

    
    solver_plugin: solver_plugins::CeresSolver
    ceres_linear_solver: SPARSE_NORMAL_CHOLESKY
    ceres_preconditioner: SCHUR_JACOBI
    ceres_threads: 4
    ceres_use_nonmonotonic_steps: false
    ceres_max_num_iterations: 200
    ceres_function_tolerance: 1e-9
    ceres_gradient_tolerance: 1e-12
    ceres_parameter_tolerance: 1e-9

    
    minimum_travel_distance: 0.1
    minimum_travel_heading: 0.05

    
    scan_buffer_size: 30
    scan_buffer_maximum_scan_distance: 20.0
    link_match_minimum_response_fine: 0.4
    link_scan_maximum_distance: 15.0
    
    
    do_loop_closing: true
    loop_search_maximum_distance: 15.0
    loop_match_minimum_chain_size: 5
    loop_match_maximum_variance_coarse: 1.0
    loop_match_minimum_response_coarse: 0.6
    loop_match_minimum_response_fine: 0.7

    
    resolution: 0.05
    max_laser_range: 30.0
    minimum_time_interval: 0.2
    
    
    transform_timeout: 1.0
    tf_buffer_duration: 30.0
    transform_publish_period: 0.02
    
    
    stack_size_to_use: 40000000
    enable_interactive_mode: false
    
    
    use_scan_matching: true
    use_scan_barycenter: true
    
    
    map_update_interval: 2.0
    
    
    throttle_scans: 1
    
    
    correlation_search_space_dimension: 0.3
    correlation_search_space_resolution: 0.005
    correlation_search_space_smear_deviation: 0.03
    
    
    fine_search_angle_offset: 0.00175
    coarse_search_angle_offset: 0.175
    coarse_angle_resolution: 0.0175
    
    
    distance_variance_penalty: 0.5
    angle_variance_penalty: 1.0
    
    
    minimum_score: 0.4
    
    
    use_response_expansion: true
